src/volatile_memory.rs

```diff
@@ -1390,6 +1434,28 @@ mod copy_slice_impl {
 
         total
     }
+
+    /// Copies `total` bytes from `slice` to `dst`
+    pub(super) unsafe fn copy_from_volatile_slice<B: BitmapSlice>(
+        dst: *mut u8,
+        slice: &VolatileSlice<'_, B>,
+        total: usize,
+    ) -> usize {
+        // SAFETY: `slice` and `dst` must be point to a contiguously allocated memory region of at
+        // least length `total`. The regions must not overlap.
+        unsafe { copy_slice(dst, slice.as_ptr(), total) }
+    }
+
+    /// Copies `total` bytes from 'src' to `slice`
+    pub(super) unsafe fn copy_to_volatile_slice<B: BitmapSlice>(
```

Comment:
I think in general, it would be better to add the safety comment on this function and then, inside the body just do  something like: `// SAFETY: guaranteed by function invariants`. Though this is just `pub(super)`, so...

Reply:
sure